ledger/src/blockstore_processor.rs

```diff
@@ -921,21 +920,54 @@ pub fn confirm_slot(
 
     let verify_transaction = {
         let bank = bank.clone();
-        move |versioned_tx: VersionedTransaction| -> Result<SanitizedTransaction> {
-            bank.verify_transaction(versioned_tx, skip_verification)
+        move |versioned_tx: VersionedTransaction,
+              verification_options: entry::EntryVerificationOptions|
+              -> Result<SanitizedTransaction> {
+            let result =
+                bank.verify_transaction(versioned_tx, verification_options.skip_verification);
+
+            match result {
+                Ok(val) => {
+                    if verification_options.verify_precompiles
+                        && verification_options.skip_verification
```

Comment:
These options are still really confusing, why would you verify only if skip verification is disabled? Maybe we need an enum for sigverify which indicates whether sigverify is totally skipped (during ledger processing) or done now with CPU or done later with GPU.

Reply:
Have added some comments and renamed some fields to hopefully clarify. Imho, I think the struct is better, since it's somewhat simpler and more elegant to use in the context of the logic start_verify_transactions rather than having a 4-branch if-statement to decide on the right enum. Having an bunch of independent flags likely also makes implementing implementing the verify function itself slightly easier, rather than needing to go through matching each enum val. So I somewhat like the struct approach better, though yes I agree it can be confusing at first, so have added some comments to hopefully clarify.

Reply:
@jstarry thanks for the suggestion. Found a relatively clean way to use an enum instead of a struct. Should hopefully be clearer now.